src/remake.jl

```diff
@@ -133,8 +120,85 @@ function remake(prob::ODEProblem; f = missing,
     end
 end
 
+"""
+    remake(prob::OptimizationProblem; f = missing, u0 = missing, p = missing,
+        lb = missing, ub = missing, int = missing, lcons = missing, ucons = missing,
+        sense = missing, kwargs = missing, _kwargs...)
+
+Remake the given `OptimizationProblem`.
+If `u0` or `p` are given as symbolic maps `ModelingToolkit.jl` has to be loaded.
+"""
+function remake(prob::OptimizationProblem;
+                f = missing,
+                u0 = missing,
+                p = missing,
+                lb = missing,
+                ub = missing,
+                int = missing,
+                lcons = missing,
+                ucons = missing,
+                sense = missing,
+                kwargs = missing,
+                _kwargs...)
+    if p === missing && u0 === missing
+        p, u0 = prob.p, prob.u0
+    else # at least one of them has a value
+        if p === missing
+            p = prob.p
+        end
+        if u0 === missing
+            u0 = prob.u0
+        end
+        if (eltype(p) <: Pair && !isempty(p)) || (eltype(u0) <: Pair && !isempty(u0)) # one is a non-empty symbolic map
+            hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :ps) ||
+                throw(ArgumentError("This problem does not support symbolic maps with `remake`, i.e. it does not have a symbolic origin." *
+                                    " Please use `remake` with the `p` keyword argument as a vector of values, paying attention to parameter order."))
+            hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :states) ||
+                throw(ArgumentError("This problem does not support symbolic maps with `remake`, i.e. it does not have a symbolic origin." *
+                                    " Please use `remake` with the `u0` keyword argument as a vector of values, paying attention to state order."))
+            p, u0 = process_p_u0_symbolic(prob, p, u0)
+        end
+    end
+
+    if f === missing
+        f = prob.f
+    end
+    if lb === missing
+        lb = prob.lb
+    end
+    if ub === missing
+        ub = prob.ub
+    end
+    if int === missing
+        int = prob.int
+    end
+    if lcons === missing
+        lcons = prob.lcons
+    end
+    if ucons === missing
+        ucons = prob.ucons
+    end
+    if sense === missing
+        sense = prob.sense
+    end
```

Comment:
This is very basic for now and not even checks anything but it can be added later.